File: 프로그래머스/2/131704. 택배상자/택배상자.js
function solution(order) {
    let answer = 0;
    
    let sub = [];
    for(let i=1; i<=order.length; i++){
        sub.push(i);
        
        while(sub.length !== 0 && sub[sub.length-1] === order[answer]){
            answer++;
            sub.pop();
        }
    }
    
    return answer;
}

File: 프로그래머스/2/131704. 택배상자/택배상자.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<int> order) {
    int answer = 0;
    stack<int> sub;
    
    int idx = 0;
    for(int i=1; i<=order.size(); i++){
        sub.push(i);
        
        while(!sub.empty() && sub.top() == order[idx]){
            idx++;
            answer++;
            sub.pop();
        }
    }
    
    return answer;
}